### DISCUSSION TITLE
Pass project structure in Assistant panel by slash command

### DISCUSSION THREAD

**Discussion by erik-balfe** [zed-industries/zed#19698] [Feature Requests]
I’d like to propose a new slash command (e.g., /tree) that displays the full project file structure in a folded spoiler. This will help users avoid clutter while giving the AI a complete view of the project.

Why It Matters:
When the AI only has a few files, it sometimes assumes those represent the entire project, leading to incorrect suggestions and frustrating experiences. A full file list would allow the assistant to accurately understand imports, ask about content in other files, and interact more naturally, especially in larger projects, where adding all files in context is pricey, difficult, or even impossible due to context window limits.

I know there’s a similar command (/project) that inserts general info about Rust projects, but I’m not sure what would be better—improving that command or creating a new one. My idea is that after pressing 'enter', my command could list the project structure and include relevant details about used libraries, like the content of package.json for JavaScript projects or equivalent files for other languages. This additional context often provides crucial insights for the LLMs to solve tasks effectively.

I’m ready to implement this myself, as it seems straightforward. I just want to ensure this idea aligns with any future plans. A simple thumbs-up or feedback would be appreciated to know it’s a good direction!

Any thoughts?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
